packages/route/src/Application/ResourceLocator/ArrayAccessObject.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Route\Application\ResourceLocator;
+
+/**
+ * A wrapper class that allows both property access (object.title) and
+ * array access (object['title']) syntax in ExpressionLanguage expressions.
+ *
+ * @implements \ArrayAccess<string, mixed>
```

Comment:
Its only used to make the expression language "easier" we should keep it internal.

```suggestion
 * A wrapper class that allows both property access (object.title) and
 * array access (object['title']) syntax in ExpressionLanguage expressions.
 *
 * @internal No backwards compatibility promises are given for this class it can be removed at any time.
 *
 * @implements \ArrayAccess<string, mixed>
```